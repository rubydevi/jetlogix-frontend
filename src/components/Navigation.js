import React, { useState, useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import {
  FaTwitter, FaFacebook, FaGooglePlus, FaPinterest, FaVimeo,
} from 'react-icons/fa';
import { useNavigation } from './NavigationContext';
import logo from '../assets/app-logo.png';

const Navigation = () => {
  const { isOpen } = useNavigation();
  const [isMobile, setIsMobile] = useState(false);

  useEffect(() => {
    const checkIfMobile = () => {
      setIsMobile(window.innerWidth <= 768);
    };

    checkIfMobile();
    window.addEventListener('resize', checkIfMobile);
    return () => {
      window.removeEventListener('resize', checkIfMobile);
    };
  }, []);

  return (
    <div
      className={`position-fixed vh-100 bg-light border-end d-flex flex-column ${(isOpen && isMobile) ? 'w-50' : 'w-0'}`}
      id="sidebar-wrapper"
    >
      <div className="sidebar-heading">
        <img alt="Logo" src={logo} className="img-fluid" style={{ width: '150px' }} />
      </div>
      <div className="list-group list-group-flush flex-grow-1">
        <NavLink exact to="/" className="list-group-item list-group-item-action">
          Home
        </NavLink>
        <NavLink to="/aeroplanes" className="list-group-item list-group-item-action">
          Private Planes
        </NavLink>
        <NavLink to="/reserve" className="list-group-item list-group-item-action">
          Reserve
        </NavLink>
        <NavLink to="/reservations" className="list-group-item list-group-item-action">
          My Reservations
        </NavLink>
        <NavLink to="/add_aeroplane" className="list-group-item list-group-item-action">
          Add Aeroplane
        </NavLink>
        <NavLink to="/delete_aeroplane" className="list-group-item list-group-item-action">
          Delete Aeroplane
        </NavLink>
        <div className="mt-auto">
          <div className="footer-icons d-flex justify-content-around mt-auto">
            <FaTwitter className="mx-2" />
            <FaFacebook className="mx-2" />
            <FaGooglePlus className="mx-2" />
            <FaPinterest className="mx-2" />
            <FaVimeo className="mx-2" />
          </div>
          <p>@2023 JetLogix</p>
        </div>
      </div>
    </div>
  );
};

export default Navigation;
