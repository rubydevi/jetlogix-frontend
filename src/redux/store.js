import { configureStore } from '@reduxjs/toolkit';
import aeroplanesReducer from './aeroplanes/aeroplanesSlice';
import reservationsReducer from './reservations/reservationSlice';

const store = configureStore({
  reducer: {
    aeroplanes: aeroplanesReducer,
    reservation: reservationsReducer,
  },
});

export default store;
