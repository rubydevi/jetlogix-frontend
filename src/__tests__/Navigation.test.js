import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Navigation from '../components/Navigation';

jest.mock('../components/NavigationContext', () => ({
  ...jest.requireActual('../components/NavigationContext'),
  useNavigation: () => ({ isOpen: false }),
}));

test('renders Navigation component correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
