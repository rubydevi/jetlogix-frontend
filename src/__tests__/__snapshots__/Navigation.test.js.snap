// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navigation component correctly 1`] = `
<div>
  <div
    class="d-block d-md-none"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light"
    >
      <div
        class="container w-100 d-flex justify-content-between align-items-center"
      >
        <button
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler pb-2"
          data-bs-target="#navbarNav"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="sidebar-heading"
        >
          <a
            aria-current="page"
            class="list-group-item navbar-brand list-group-item-action  active"
            href="/"
          >
            <img
              alt="Logo"
              class="img-fluid"
              src="app-logo.png"
              style="width: 80px;"
            />
          </a>
        </div>
        <div
          class="collapse navbar-collapse"
          id="navbarNav"
        />
      </div>
    </nav>
  </div>
  <div
    class="d-none d-md-block"
  >
    <div
      class="position-absolute top-0 start-0 vh-100 bg-light border-end d-flex flex-column w-0"
      id="sidebar-wrapper"
    >
      <div
        class="sidebar-heading ps-4"
      >
        <img
          alt="Logo"
          class="img-fluid"
          src="app-logo.png"
          style="width: 150px;"
        />
      </div>
      <div
        class="mt-auto"
      >
        <div
          class="footer-icons d-flex justify-content-around mt-auto"
        >
          <svg
            class="mx-2 text-color"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
            />
          </svg>
          <svg
            class="mx-2 text-color "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
            />
          </svg>
          <svg
            class="mx-2 text-color "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256,8C119.1,8,8,119.1,8,256S119.1,504,256,504,504,392.9,504,256,392.9,8,256,8ZM185.3,380a124,124,0,0,1,0-248c31.3,0,60.1,11,83,32.3l-33.6,32.6c-13.2-12.9-31.3-19.1-49.4-19.1-42.9,0-77.2,35.5-77.2,78.1S142.3,334,185.3,334c32.6,0,64.9-19.1,70.1-53.3H185.3V238.1H302.2a109.2,109.2,0,0,1,1.9,20.7c0,70.8-47.5,121.2-118.8,121.2ZM415.5,273.8v35.5H380V273.8H344.5V238.3H380V202.8h35.5v35.5h35.2v35.5Z"
            />
          </svg>
          <svg
            class="mx-2 text-color "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M496 256c0 137-111 248-248 248-25.6 0-50.2-3.9-73.4-11.1 10.1-16.5 25.2-43.5 30.8-65 3-11.6 15.4-59 15.4-59 8.1 15.4 31.7 28.5 56.8 28.5 74.8 0 128.7-68.8 128.7-154.3 0-81.9-66.9-143.2-152.9-143.2-107 0-163.9 71.8-163.9 150.1 0 36.4 19.4 81.7 50.3 96.1 4.7 2.2 7.2 1.2 8.3-3.3.8-3.4 5-20.3 6.9-28.1.6-2.5.3-4.7-1.7-7.1-10.1-12.5-18.3-35.3-18.3-56.6 0-54.7 41.4-107.6 112-107.6 60.9 0 103.6 41.5 103.6 100.9 0 67.1-33.9 113.6-78 113.6-24.3 0-42.6-20.1-36.7-44.8 7-29.5 20.5-61.3 20.5-82.6 0-19-10.2-34.9-31.4-34.9-24.9 0-44.9 25.7-44.9 60.2 0 22 7.4 36.8 7.4 36.8s-24.5 103.8-29 123.2c-5 21.4-3 51.6-.9 71.2C65.4 450.9 0 361.1 0 256 0 119 111 8 248 8s248 111 248 248z"
            />
          </svg>
          <svg
            class="mx-2 text-color "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M403.2 32H44.8C20.1 32 0 52.1 0 76.8v358.4C0 459.9 20.1 480 44.8 480h358.4c24.7 0 44.8-20.1 44.8-44.8V76.8c0-24.7-20.1-44.8-44.8-44.8zM377 180.8c-1.4 31.5-23.4 74.7-66 129.4-44 57.2-81.3 85.8-111.7 85.8-18.9 0-34.8-17.4-47.9-52.3-25.5-93.3-36.4-148-57.4-148-2.4 0-10.9 5.1-25.4 15.2l-15.2-19.6c37.3-32.8 72.9-69.2 95.2-71.2 25.2-2.4 40.7 14.8 46.5 51.7 20.7 131.2 29.9 151 67.6 91.6 13.5-21.4 20.8-37.7 21.8-48.9 3.5-33.2-25.9-30.9-45.8-22.4 15.9-52.1 46.3-77.4 91.2-76 33.3.9 49 22.5 47.1 64.7z"
            />
          </svg>
        </div>
        <p
          class="text-muted small text-center py-2"
        >
          @2023 JetLogix
        </p>
      </div>
    </div>
  </div>
</div>
`;
